snap/quota/quota.go

```diff
@@ -264,18 +265,67 @@ func maxq(a, b quantity.Size) quantity.Size {
 	return b
 }
 
-func (grp *Group) getAllowedCPUs() []int {
+// GetLocalCPUSetQuota returns the current CPU set quota for the group. This
+// does not return any inheritted CPU set quota.
+func (grp *Group) GetLocalCPUSetQuota() []int {
 	if grp.CPULimit == nil || len(grp.CPULimit.AllowedCPUs) == 0 {
 		return []int{}
 	}
 	return grp.CPULimit.AllowedCPUs
 }
 
-func (grp *Group) getTotalCPUPercentage() int {
+// GetCPUSetQuota returns the currently active CPU set quota for this group, which
+// includes the case where the CPU set is inherited from a parent group.
+func (grp *Group) GetCPUSetQuota() []int {
+	localCPUSet := grp.GetLocalCPUSetQuota()
+	if len(localCPUSet) != 0 {
+		return localCPUSet
+	}
+
+	parent := grp.parentGroup
+	for parent != nil {
+		if parent.CPULimit != nil && len(parent.CPULimit.AllowedCPUs) != 0 {
+			return parent.CPULimit.AllowedCPUs
+		}
+		parent = parent.parentGroup
+	}
+	return nil
+}
+
+// export it for test
+var runtimeNumCPU = runtime.NumCPU
```

Comment:
nitpick: I would move all the exported methods to the top of the file

Reply:
makes sense!